git/repo/base.py

```diff
@@ -296,6 +296,16 @@ def __init__(
         else:
             self.odb = odbt(rootpath)
 
+        # Check dubious ownership
+        try:
+            if not (self.bare or Git.is_cygwin()):
+                _ = self.git.status()
```

Comment:
`git status` can be a very expensive operation, and even when it's not it's still expensive and can't be happening when creating a new Repository instance. Interestingly, it will only be cheap if it fails due to dubious ownership.

Can you find another invocation that ideally just checks for permissions and nothing else?

Reply:
You came to the same dilemma that I get when I was implementing this...
I thought to just keep it as message enhancement when the git command gets executed but then I thought that it could be little late for the user that already generated the repo instance and maybe did some other stuff but I didn't consider that it could be very expensive operation, not sure if do a manual check could be less expensive just like git is doing it, check if the ownership is different to the actual user but then a lot of extra logic will need to be implemented to consider the git config file exceptions, etc. Kind of duplicated logic and more conde to maintain.

Is it an easy way to check how expensive it is to execute this command? it should be just one time at the repo instantiation, maybe change it for another operation less expensive like check the latest commit?

Reply:
I thought about this and realized that adding something into the instantiation-path would still be unacceptable even if it could happen at no cost. The reason for this is that right now, some operations can be performed without `git` and safely so, so users shouldn't be held back by something that `git` *would* say. Further, doing so would be a breaking change.

Personally I think `git` was 'adamant' in their implementation as they put defence of a particular issue before usability. It's understandable, but also far from the best possible implementation of such mitigation.

`gitoxide` was lucky enough as second-mover to have a fresh take on this, as it will simply ignore configuration data from untrusted sources. GitPython should do the same, as much as possible, and prefer to provide better error messages if `git` fails.

Reply:
get it! now I see what is the philosophy of GitPython, let me reformulate this implementation and go with an enhancement of the message in the cases where it apply.